cogs/admin/ban.py

```diff
@@ -1,56 +1,93 @@
+        reason=Param(None, desc="Provide a reason"),
+    ):
+        embed = disnake.Embed(
+            description=(
+                f"✅ **{user.display_name}#{user.discriminator} is banned for {reason}"
+            ),
+            color=disnake.Color.red(),
+        )
+        await inter.channel.send(embed=embed)
+        await user.ban(user, reason=reason)
+
+    @commands.slash_command(description="Unban user from the Server.")
+    async def unban(
+        self,
+        inter: disnake.ApplicationCommandInteraction,
+        user: disnake.User = Param(None, desc="Specify a user"),
+        reason: str = Param(None, desc="Provide a reason"),
+    ):
+        embed = disnake.Embed(
+            description=(
+                f"✅ **{user.display_name}#{user.discriminator} had been unbanned"
+            ),
+            color=disnake.Color.green(),
+        )
+        await inter.channel.send(embed=embed)
+        await user.unban()
+
+    @commands.slash_command(description="Tempban user from the server.")
+    async def tempban(
+        self,
+        inter: disnake.ApplicationCommandInteraction,
+        user: disnake.User = Param(None, desc="Specify a user"),
+        time: str = Param(None, desc="Specify a time"),
+        reason: str = Param(None, desc="Provide a reason"),
+    ):
+        time_convert = {"s": 1, "m": 60, "h": 3600, "d": 86400}
+        ban_time = int(time[0]) * time_convert[time[-1]]
+        end_time = datetime.now() + timedelta(seconds=ban_time)
+        embed = disnake.Embed(
+            description=(
+                f"✅ **{user.display_name}#{user.discriminator} had been tempban for {time} successfully**"
+            ),
+            color=disnake.Color.green(),
+        )
+
+        conn = create_connection(r"sqlite.db")
+        sql = f" INSERT INTO temp_ban(guild_id,user_id,end_time) VALUES({inter.guild.id},{user.id},{end_time.timestamp()}) "
+        cur = conn.cursor()
+        cur.execute(sql)
+        conn.commit()
+        await user.ban(reason=reason)
+        await inter.response.send_message(embed=embed)
+        # await asyncio.sleep(ban_time)
+        # await user.unban(user)
+
+    # TODO change this to how often you want this to run, i'd suggest every minute
+    @tasks.loop(minutes=1.0)
+    async def temp_ban_checker(self):
+        conn = create_connection(r"sqlite.db")
+        timestamp_now = datetime.now().timestamp()
+        cur = conn.cursor()
+        cur.execute(f"SELECT * FROM temp_ban WHERE end_time < {timestamp_now}")
+        rows = cur.fetchall()
+        for row in rows:
+            selected_guild = self.bot.get_guild(row[1])
+            bans = await selected_guild.bans()
+            for ban in bans:
+                if row[2] == ban.user.id:
+                    print (f"unbanning {ban.user}")
+                    await selected_guild.unban(ban.user)
+
+    @temp_ban_checker.before_loop
+    async def before_printer(self):
```

Comment:
This just makes sure we dont start the un-banning checker until the bot is fully loaded